x/src/cargo.rs

```diff
@@ -86,39 +107,65 @@ impl Cargo {
         self
     }
 
+    // Passes extra environment variables to be passed to the executable.
+    pub fn env_additions<I, S>(&mut self, envs: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (S, S)>,
+        S: AsRef<OsStr>,
+        //        S: std::convert::AsRef<std::ffi::OsStr>,
+    {
+        self.inner.envs(envs);
+        self
+    }
+
     pub fn run(&mut self) -> Result<()> {
         self.inner.stdout(Stdio::inherit()).stderr(Stdio::inherit());
         self.do_run(true).map(|_| ())
     }
 
+    // This execution will in general not produce std out file output, but will
+    // instead return a byte Vec containing the execution's stdout messages.
     pub fn run_with_output(&mut self) -> Result<Vec<u8>> {
         self.inner.stderr(Stdio::inherit());
         // Since system out hijacked don't log for this command
         self.do_run(false).map(|o| o.stdout)
     }
 
+    // Internal run command, where the magic happens
+    // * timing information can be is captured.
+    // * command information necessary to rerun the command is logged.
     fn do_run(&mut self, log: bool) -> Result<Output> {
+        //this are arguments are passed through cargo to underlaying
+        //executable
```

Comment:
Bad formatting of the comment and improper english

Reply:
fixed and other comments checked.